populus/project.py

```diff
@@ -1,84 +1,103 @@
 import os
+import shutil
 import itertools
 import warnings
 
+from eth_utils import (
+    to_tuple,
+)
+
 from populus.compilation import (
     compile_project_contracts,
 )
+
 from populus.config import (
     ChainConfig,
     CompilerConfig,
     Config,
-    get_default_config_path,
     load_config as _load_config,
+    load_user_config,
     load_config_schema,
     write_config as _write_config,
 )
+
+from populus.config.defaults import (
+    get_default_config_path,
+)
+
 from populus.utils.chains import (
     get_base_blockchain_storage_dir,
 )
+
 from populus.utils.compile import (
     get_build_asset_dir,
     get_compiled_contracts_asset_path,
     get_contracts_source_dirs,
     get_project_source_paths,
     get_test_source_paths,
 )
+
 from populus.utils.filesystem import (
-    relpath,
     get_latest_mtime,
 )
-from populus.utils.config import (
-    check_if_json_config_file_exists,
-    get_default_project_config_file_path,
+
+from populus.config.helpers import (
     get_json_config_file_path,
 )
+
 from populus.utils.testing import (
     get_tests_dir,
 )
 
 
 class Project(object):
-    def __init__(self, config_file_path=None):
-        self.config_file_path = config_file_path
+
+    project_dir = None
+    config_file_path = None
+
+    def __init__(self, project_dir=None, user_config=None, create_config_file=False):
```

Comment:
Can you explain your reasoning behind having this new `create_config_file` argument to the `Project` constructor?

Reply:
assuming populus cli points to an existing project unless explicitly creating new one

Reply:
Right, but why would the project object be responsible for writing the default config file.  That seems odd enough that I question whether it's the correct place.  Seems like something that would happen *above* this level.